api/envoy/type/matcher/v3/node.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package envoy.type.matcher.v3;
+
+import "envoy/type/matcher/v3/metadata.proto";
+import "envoy/type/matcher/v3/string.proto";
+
+import "udpa/annotations/versioning.proto";
+
+option java_package = "io.envoyproxy.envoy.type.matcher.v3";
+option java_outer_classname = "NodeProto";
+option java_multiple_files = true;
+
+// [#protodoc-title:NodeMatcher]
+
+// Specifies the way to match a Node.
+message NodeMatcher {
+  option (udpa.annotations.versioning).previous_message_type = "envoy.type.matcher.NodeMatcher";
+
+  // Specifies match criteria on the node id.
+  StringMatcher node_id = 1;
+
+  // Specifies match criteria on the node metadata.
+  repeated MetadataMatcher node_metadatas = 2;
```

Comment:
Is this `AND` semantics? Might it be the case that someone one day wants `OR`?

Reply:
Yes, this is intended to be AND. If OR is needed, one can have repeated NodeMatcher, and only include either node_id or node_metadatas in them.

Reply:
Can you clarify this in the docs per my other comment both here and there?

Reply:
Done. Thank you.